src/Interpreters/convertFieldToType.cpp

```diff
@@ -190,9 +190,14 @@ Field convertFieldToTypeImpl(const Field & src, const IDataType & type, const ID
         if (which_type.isInt256()) return convertNumericType<Int256>(src, type);
         if (which_type.isFloat32()) return convertNumericType<Float32>(src, type);
         if (which_type.isFloat64()) return convertNumericType<Float64>(src, type);
-        if (const auto * ptype = typeid_cast<const DataTypeDecimal<Decimal32> *>(&type)) return convertDecimalType(src, *ptype);
-        if (const auto * ptype = typeid_cast<const DataTypeDecimal<Decimal64> *>(&type)) return convertDecimalType(src, *ptype);
-        if (const auto * ptype = typeid_cast<const DataTypeDecimal<Decimal128> *>(&type)) return convertDecimalType(src, *ptype);
+#define M(TYPE) \
+        if (const auto * ptype = typeid_cast<const DataTypeDecimal<TYPE> *>(&type)) return convertDecimalType(src, *ptype);
+
+        M(Decimal32)
+        M(Decimal64)
+        M(Decimal128)
+        M(Decimal256)
+#undef M
```

Comment:
Let's leave a previous version before force push when just 1 line was added with the `if` for Decimal256.

Reply:
OK, I have fixed it.